testing/test_nose.py

```diff
@@ -211,6 +211,96 @@ def test_world():
     result.stdout.fnmatch_lines(["*2 passed*"])
 
 
+def test_fixtures_nose_setup_module_issue8394(pytester: Pytester) -> None:
```

Comment:
I'm thinking we can combine all of the new tests in this file into a single test which contains all of the functions/methods/classmethods, I think it will be sufficient for the regression test.